infrastructure/provisioning/terraform/cloud-sql.tf

```
@@ -0,0 +1,117 @@
+resource "google_sql_database_instance" "master" {
+  provider = "google-beta"
+
+  name   = "material-list-alpha2"
+  region = "${var.region}"
+
+  database_version = "MYSQL_5_7"
+
+  depends_on = [
+    "google_service_networking_connection.private_vpc_connection"
+  ]
+  settings {
+    # Second-generation instance tiers are based on the machine
+    # type. See argument reference below.
+    tier            = "${var.db_instance_type}"
+    disk_autoresize = true
+    disk_size       = 50
+    ip_configuration {
+      ipv4_enabled    = false
+      private_network = "${google_compute_network.private_network.self_link}"
+    }
+
+    backup_configuration {
+      binary_log_enabled = "true"
+      enabled            = "true"
+      start_time         = "03:04"
+    }
+
+    location_preference {
+      zone = "${var.zone_1}"
+    }
+  }
+}
+
+resource "google_sql_database_instance" "replica" {
+  name   = "material-list-beta2"
+  region = "${var.region}"
+
+  database_version     = "MYSQL_5_7"  
+  master_instance_name = "${google_sql_database_instance.master.name}"
```

I thought this syntax was deprecated with 0.12? https://www.terraform.io/upgrade-guides/0-12.html#first-class-expressions

```suggestion
  master_instance_name = google_sql_database_instance.master.name
```

---

huh ... I might have quite a lot of .tf files I need to rework.